contrib/platform/src/com/sun/jna/platform/win32/WinspoolUtil.java

```diff
@@ -45,6 +46,62 @@
 
         return (PRINTER_INFO_1[]) pPrinterEnum.toArray(pcReturned.getValue());
     }
+    
+    public static PRINTER_INFO_2[] getPrinterInfo2() {
+		return getPrinterInfo2(Winspool.PRINTER_ENUM_LOCAL);
+	}
+	
+	/**
+	 * Returns printers that are physically attached to the local machine as
+	 * well as remote printers to which it has a network connection.
+	 */
+	public static PRINTER_INFO_2[] getAllPrinterInfo2() {
+		// When Name is NULL, setting Flags to PRINTER_ENUM_LOCAL | PRINTER_ENUM_CONNECTIONS
+		// enumerates printers that are installed on the local machine.
+		// These printers include those that are physically attached to the local machine 
+		// as well as remote printers to which it has a network connection.
+		// See https://msdn.microsoft.com/en-us/library/windows/desktop/dd162692(v=vs.85).aspx
+		return getPrinterInfo2(Winspool.PRINTER_ENUM_LOCAL | Winspool.PRINTER_ENUM_CONNECTIONS);
+	}
+	
+	private static PRINTER_INFO_2[] getPrinterInfo2(int flags) {
+		IntByReference pcbNeeded = new IntByReference();
+		IntByReference pcReturned = new IntByReference();
+		Winspool.INSTANCE.EnumPrinters(flags, null, 2, null, 0, pcbNeeded, pcReturned);
```

Comment:
Failures anywhere here need to raise `Win32Exception`, check explicitly for whatever the return code is that tells you to call the function again with an actual buffer.